spec/requests/post_request_spec.rb

```diff
@@ -0,0 +1,31 @@
+require 'rails_helper'
+
+RSpec.describe 'Posts', type: :request do
+  describe 'GET /posts' do
+    it 'returns http success' do
+      get '/posts'
+      # test for http response status
+      expect(response).to have_http_status(:success)
+      expect(response.status).to eq(200)
+      # correct response is rendered
+      expect(response).to render_template(:index)
+      # include correct placeholders
+      expect(response.body).to include('All posts')
+      expect(response.body).to include('Hello')
+      expect(response.body).to include('This is the second post!!')
```

Comment:
**[OPTIONAL]**

- [ ] It would be nice if you make separate it blocks for each expect 👍 because the whole purpose of testing is to divide your whole app into separate test blocks so in the future if your app shows unexpected behavior then you can run these tests that the error is coming from which code block because your app won't be a small app it will be so much bigger or even if its not big it doesn't mean that finding bug would be easy then so these test cases will help you a lot 👍 because if the test will fail then it will show up with the test case that this test case is failing 🤔